integrationTests/tesGameHelperFunctions.go

```diff
@@ -0,0 +1,259 @@
+	tx := &transaction.Transaction{
+		Nonce:    0,
+		RcvAddr:  scAddress,
+		SndAddr:  twa.PkTxSignBytes,
+		Data:     fmt.Sprintf("endGame@%d", round),
+		GasPrice: 0,
+		GasLimit: 1000000000000,
+	}
+	txBuff, _ := TestMarshalizer.Marshal(tx)
+	tx.Signature, _ = twa.SingleSigner.Sign(twa.SkTxSign, txBuff)
+
+	fmt.Printf("End %s\n", hex.EncodeToString(twa.PkTxSignBytes))
+
+	return tx
+}
+
+func createTxJoinGame(twa *TestWalletAccount, joinGameVal *big.Int, round int, scAddress []byte) *transaction.Transaction {
+	tx := &transaction.Transaction{
+		Nonce:    0,
+		Value:    joinGameVal,
+		RcvAddr:  scAddress,
+		SndAddr:  twa.Address.Bytes(),
+		Data:     fmt.Sprintf("joinGame@%d", round),
+		GasPrice: 0,
+		GasLimit: 1000000000000,
+	}
+	txBuff, _ := TestMarshalizer.Marshal(tx)
+	tx.Signature, _ = twa.SingleSigner.Sign(twa.SkTxSign, txBuff)
+
+	fmt.Printf("Join %s\n", hex.EncodeToString(twa.Address.Bytes()))
+
+	return tx
+}
+
+// PlayerJoinsGame creates and sends a join game transaction to the SC
+func PlayerJoinsGame(
+	nodes []*TestProcessorNode,
+	player *TestWalletAccount,
+	joinGameVal *big.Int,
+	round int,
+	scAddress []byte,
+) {
+	txDispatcherNode := getNodeWithinSameShardAsPlayer(nodes, player.Address.Bytes())
+	fmt.Println("Calling SC.joinGame...")
+	txScCall := createTxJoinGame(player, joinGameVal, round, scAddress)
+	txDispatcherNode.SendTransaction(txScCall)
+}
+
+// NodeEndGame creates and sends an end game transaction to the SC
+func NodeEndGame(
+	nodes []*TestProcessorNode,
+	idxNode int,
+	round int,
+	scAddress []byte,
+) {
+
+	fmt.Println("Calling SC.endGame...")
+	txScCall := createTxEndGame(nodes[idxNode].OwnAccount, round, scAddress)
+	nodes[idxNode].SendTransaction(txScCall)
+	time.Sleep(stepDelay)
+
+	fmt.Println(MakeDisplayTable(nodes))
+}
+
+func createTxRewardAndSendToWallet(
+	tnOwner *TestWalletAccount,
+	winnerAddress []byte,
+	prizeVal *big.Int,
+	round int,
+	scAddress []byte,
+) *transaction.Transaction {
+	tx := &transaction.Transaction{
+		Nonce:    0,
+		Value:    big.NewInt(0),
+		RcvAddr:  scAddress,
+		SndAddr:  tnOwner.PkTxSignBytes,
+		Data:     fmt.Sprintf("rewardAndSendToWallet@%d@%s@%X", round, hex.EncodeToString(winnerAddress), prizeVal),
+		GasPrice: 0,
+		GasLimit: 100000,
+	}
+	txBuff, _ := TestMarshalizer.Marshal(tx)
+	tx.Signature, _ = tnOwner.SingleSigner.Sign(tnOwner.SkTxSign, txBuff)
+
+	fmt.Printf("Reward %s\n", hex.EncodeToString(winnerAddress))
+
+	return tx
+}
+
+// NodeCallsRewardAndSend - smart contract owner sends reward transaction
+func NodeCallsRewardAndSend(
+	nodes []*TestProcessorNode,
+	idxNodeOwner int,
+	winnerAddress []byte,
+	prize *big.Int,
+	round int,
+	scAddress []byte,
+) {
+
+	fmt.Println("Calling SC.rewardAndSendToWallet...")
+	txScCall := createTxRewardAndSendToWallet(nodes[idxNodeOwner].OwnAccount, winnerAddress, prize, round, scAddress)
+	nodes[idxNodeOwner].SendTransaction(txScCall)
```

Comment:
a time.Sleep is required also here. Might add also fmt.Println(MakeDisplayTable(nodes))

Reply:
done